repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/servlet/http/MutableHttpServletResponse.java

```diff
@@ -302,11 +302,16 @@ public void sendError(int code) throws IOException {
 
     @Override
     public void sendError(int code, String message) {
-        super.setStatus(code);
-        this.message = message;
+        setStatus(code, message);
         error = true;
     }
 
+    @Override
+    public void setStatus(int code, String message) {
+        super.setStatus(code, message);
+        this.message = message;
```

Comment:
I expect the reason for sendError calling super.setStatus rather than super.sendError is that super.sendError commits the output stream, which we defer until the end with our wrapper. So using setStatus is probably the right call.

Even so, the second parameter of setStatus is not the same thing as the second parameter of sendError. Conflating the two leads to very non-obvious behavior, in my opinion.

Yes, the ambiguity is what is biting us. Which seems like a reason not to use it if we don't have to.

Your point about the ResponseMessagingService is true, but it seems like this is the wrong fix for it. In general, we don't write the status message. We send error messages as response bodies. For one, the error message may be fairly verbose, and so it doesn't belong on the status line. The fact of the matter is that delegating filters are trying to communicate some error. We shouldn't squelch that error, nor should we stick it in the status line. Perhaps a better alternative would be to send empty messages when delegating and a message is not beneficial. That was IF_EMPTY would still catch the response.